env_vars_loader.bzl

```diff
@@ -4,7 +4,7 @@ def _impl(repository_ctx):
   env_vars = repository_ctx.attr.env_vars
   bzl_vars = ""
   for env_var in env_vars:
-    bzl_var = repository_ctx.execute(["printenv", env_var]).stdout.rstrip()
+    bzl_var = repository_ctx.getenv(env_var)
```

Comment:
what does this do? (i.e. wdym by "register as used"?)

Reply:
Bazel constructs a cache key for repository rules.  Passing an environment variable name to `getenv` or to the `environ` attribute of `repository_rule` will add the hash of that environment variable to the cache key.
Just executing `printenv` like this would not put anything into the cache key, potentially allowing the repository to be vendored across boundaries where the environment variable has changed.